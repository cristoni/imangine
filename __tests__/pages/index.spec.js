import React from 'react';
import Index from '../../pages/index';

test('Index renders the main page', () => {
  const wrapper = shallow(<Index />);
  expect(wrapper).toMatchSnapshot();

  expect(wrapper.state().files).toEqual([]);

  const mockAcceptedFiles = ['file', 'file', 'file'];
  const mockRejectedFiles = [];

  wrapper.instance().onDropHandler(mockAcceptedFiles, mockRejectedFiles);

  expect(wrapper.state().files).toEqual(mockAcceptedFiles);
  expect(wrapper).toMatchSnapshot();
});