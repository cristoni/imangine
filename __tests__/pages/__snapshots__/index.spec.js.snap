// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders the main page 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <p>
        ImaNgine allow you to optimize your images for free.
        <br />
        It is developed using 
        <a
          href="https://reactjs.org/"
          target="_blank"
        >
          React
        </a>
        , more specifically 
        <a
          href="https://nextjs.org/"
          target="_blank"
        >
          Next.js
        </a>
        . You can find it's source code 
        <a
          href="https://github.com/cristoni/imangine"
          target="_blank"
        >
          here
        </a>
        .
      </p>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h5
        className="mb-2 mt-4"
      >
        Select the 
        <span
          className="hl"
        >
          Images
        </span>
         that you want to process
      </h5>
      <Uploader
        onDropHandler={[Function]}
      />
    </div>
  </div>
  <SideModal
    close={[Function]}
    open={false}
  >
    <ProcessConfiguration
      files={Array []}
    />
  </SideModal>
</div>
`;

exports[`Index renders the main page 2`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <p>
        ImaNgine allow you to optimize your images for free.
        <br />
        It is developed using 
        <a
          href="https://reactjs.org/"
          target="_blank"
        >
          React
        </a>
        , more specifically 
        <a
          href="https://nextjs.org/"
          target="_blank"
        >
          Next.js
        </a>
        . You can find it's source code 
        <a
          href="https://github.com/cristoni/imangine"
          target="_blank"
        >
          here
        </a>
        .
      </p>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h5
        className="mb-2 mt-4"
      >
        Select the 
        <span
          className="hl"
        >
          Images
        </span>
         that you want to process
      </h5>
      <Uploader
        onDropHandler={[Function]}
      />
      <button
        className="btn btn-warning btn-lg shadow d-flex justify-content-center ml-auto mr-auto mt-4"
        onClick={[Function]}
      >
        Process selected files 
        <i
          className="ml-2 material-icons"
        >
          cloud_upload
        </i>
      </button>
    </div>
    <div
      className="col-auto mw-50"
    >
      <h5
        className="mb-2 mt-4"
      >
        Selected 
        <span
          className="hl"
        >
          I
        </span>
        mages
      </h5>
      <FilesList
        files={
          Array [
            "file",
            "file",
            "file",
          ]
        }
      />
    </div>
  </div>
  <SideModal
    close={[Function]}
    open={false}
  >
    <ProcessConfiguration
      files={
        Array [
          "file",
          "file",
          "file",
        ]
      }
    />
  </SideModal>
</div>
`;
